src/app/shared/theme-picker/theme-picker.ts

```diff
@@ -71,8 +70,12 @@ export class ThemePicker implements OnInit, OnDestroy {
 
   ngOnInit() {
     this._queryParamSubscription = this._activatedRoute.queryParamMap
-      .pipe(map((params: ParamMap) => params.get('theme')), filter(Boolean))
-      .subscribe((themeName: string) => this.installTheme(themeName));
+      .pipe(map((params: ParamMap) => params.get('theme')))
```

Comment:
I think the previous approach was valid, you just needed to do `themeName!` since it's guaranteed to be defined.

Reply:
```ts
    this._queryParamSubscription = this._activatedRoute.queryParamMap
    .pipe(map((params: ParamMap) => params.get('theme')), filter(Boolean))
    .subscribe((themeName: string) => this.installTheme(themeName!));
```
Results in:
```ts
ERROR in src/app/shared/theme-picker/theme-picker.ts:74:16 - error TS2769: No overload matches this call.
  Overload 1 of 5, '(observer?: NextObserver<unknown> | ErrorObserver<unknown> | CompletionObserver<unknown> | undefined): Subscription', gave the following error.
    Argument of type '(themeName: string) => void' is not assignable to parameter of type 'NextObserver<unknown> | ErrorObserver<unknown> | CompletionObserver<unknown> | undefined'.
      Property 'complete' is missing in type '(themeName: string) => void' but required in type 'CompletionObserver<unknown>'.
  Overload 2 of 5, '(next?: ((value: unknown) => void) | undefined, error?: ((error: any) => void) | undefined, complete?: (() => void) | undefined): Subscription', gave the following error.
    Argument of type '(themeName: string) => void' is not assignable to parameter of type '(value: unknown) => void'.
      Types of parameters 'themeName' and 'value' are incompatible.
        Type 'unknown' is not assignable to type 'string'.

74     .subscribe((themeName: string) => this.installTheme(themeName!));
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  node_modules/rxjs/internal/types.d.ts:64:5
    64     complete: () => void;
           ~~~~~~~~
    'complete' is declared here.
```
This compiles, but is not great...
```ts
.subscribe((themeName: any) => this.installTheme(themeName));
```